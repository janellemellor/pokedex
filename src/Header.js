import React, { Component } from "react";

export default class Header extends Component {
    render() {
        return (
            <header>
                <h2>Pokedex</h2>
            </header>
        );
    }
}