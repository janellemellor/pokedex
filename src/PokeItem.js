import React, { Component } from "react";

export default class PokeItem extends Component {
    render() {
        return (
            <li>
                <div>
                    <h3>{this.props.title}</h3>
                    <img src={this.props.img} alt=''></img>    
                    <p>{this.props.type1}</p>
                    <p>{this.props.type2}</p>
                    <p>{this.props.ability1}</p>
                    <p>{this.props.ability2}</p>
                </div>    
            </li>
      
        );
    }
}