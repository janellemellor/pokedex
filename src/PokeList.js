import React, { Component } from "react";
import PokeItem from './PokeItem.js';

export default class PokeList extends Component {
    render() {
        return <ul>
            {this.props.pokemon.map(poke => <PokeItem title={poke.pokemon} img={poke.url_image} type1={poke.type_1}type2={poke.type_2} ability1={poke.ability_1} ability2={poke.ability_2}/>
            )}

        </ul>
        
    }
}

