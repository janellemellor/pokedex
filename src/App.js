import React, { Component } from 'react';
import './App.css';
import Header from './Header.js';
import PokeList from './PokeList.js';
import Paging from './Paging.js';
import SearchPoke from './SearchPoke.js';
import request from 'superagent';


export default class App extends Component {
state = { 
  pokemon: [],
  search: ''
};

  async componentDidMount() {
    const pokeData = await request.get('https://alchemy-pokedex.herokuapp.com/api/pokedex');
   this.setState({ pokemon: pokeData.body.results })
   console.log(this.state.poke);
  }  
  render() {
  
  return (
    <div>
      <Header />
      <SearchPoke />
      <PokeList pokemon={this.state.pokemon} />
        {/* // ? this.state.pokemon : 'loading'} /> */}
    </div>
  );
}}


