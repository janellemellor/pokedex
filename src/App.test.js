import React from 'react';
import App from './App.js';
import Header from './Header.js';
import PokeList from './PokeList.js';
import PokeItem from './PokeItem.js';
import renderer from 'react-test-renderer';

test('renders App.js correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Header.js correctly', () => {
  const tree = renderer
  .create(<Header />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders PokeList.js correctly', () => {
  const tree = renderer
  .create(<PokeList pokemon={this.state.pokemon} />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders PokeItem.js correctly', () => {
  const tree = renderer
  .create(<PokeItem />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});