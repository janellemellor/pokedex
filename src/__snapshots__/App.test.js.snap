// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App.js correctly 1`] = `
<div>
  <header>
    <h2>
      Pokedex
    </h2>
  </header>
  <ul />
</div>
`;
